src/utils/binding.jl

```diff
@@ -0,0 +1,102 @@
+"""
+Heuristic for showing completions.  A binding is relevant when all are true:
+- it isn't generated by the compiler
+- if nonglobal, it's defined before the cursor
+- (if global) it doesn't contain or immediately precede the cursor
+"""
+function is_relevant(ctx::JL.AbstractLoweringContext,
+                     binding::JL.BindingInfo,
+                     cursor::Int)
+    (;start, stop) = JS.byte_range(JL.binding_ex(ctx, binding.id))
+    !binding.is_internal &&
+        !in(cursor, start:(stop+1)) &&
+        (binding.kind === :global
+         # || we could relax this for locals defined before the end of the
+         #    largest for/while containing the cursor
+         || cursor > start)
+end
+
+let lowering_module = Module()
+    global function jl_lower_for_completion(st0)
```

Comment:
```suggestion
    global function jl_lower_for_scope_resolution(st0)
```
Maybe we should rename this function now.

Reply:
I agree. resolved  f16724a